sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sims-to-sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -166,6 +229,455 @@ describe(describeProcessorRootTest(QueueNames.SIMSToSFASIntegration), () => {
+          restrictionType: RestrictionType.Federal,
+        },
+      });
+      const resolvedRestriction = await saveFakeStudentRestriction(
+        db.dataSource,
+        {
+          student,
+          application,
+          restriction: provincialRestriction,
+        },
+        {
+          isActive: false,
+        },
+      );
+      await db.studentRestriction.update(
+        {
+          id: resolvedRestriction.id,
+        },
+        {
+          updatedAt: simsDataUpdatedDate,
+        },
+      );
+
+      //Create a restriction for federal which should not be sent.
+      const restriction = await saveFakeStudentRestriction(
+        db.dataSource,
+        {
+          student,
+          application,
+          restriction: federalRestriction,
+        },
+        {
+          isActive: false,
+        },
+      );
+
+      await db.studentRestriction.update(
+        {
+          id: restriction.id,
+        },
+        {
+          updatedAt: simsDataUpdatedDate,
+        },
+      );
+
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Mock the current date.
+      MockDate.set(mockedCurrentDate);
+
+      // Expected file name.
+      const expectedFileName = buildExpectedFileName(mockedCurrentDate);
+
+      // Act
+      const processingResult = await processor.generateSFASBridgeFile(
+        mockedJob.job,
+      );
+
+      // Assert
+      // Assert process result.
+      expect(processingResult).toEqual([
+        "Process finalized with success.",
+        "Student records sent: 1.",
+        "Application records sent: 0.",
+        "Restriction records sent: 1.",
+        `Uploaded file name: ${expectedFileName}.`,
+      ]);
+      expect(
+        mockedJob.containLogMessages([
+          `Processing data since ${latestBridgeFileDate} until ${mockedCurrentDate}.`,
+          "Found 1 student(s) with updates.",
+          `SIMS to SFAS file ${expectedFileName} has been uploaded successfully.`,
+          `SIMS to SFAS file log has been created with file name ${expectedFileName} and reference date ${mockedCurrentDate}.`,
+        ]),
+      ).toBe(true);
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const [header, studentRecord, restrictionRecord, footer] =
+        uploadedFile.fileLines;
+      expect(header).toBe(buildHeader(mockedCurrentDate));
+      expect(footer).toBe("999000000002");
+      expect(studentRecord).toBe(buildStudentRecord(student));
+      expect(restrictionRecord).toBe(
+        buildRestrictionRecord(resolvedRestriction),
+      );
+      // Check the database for creation of SFAS bridge log.
+      const uploadedFileLog = await db.sfasBridgeLog.existsBy({
+        generatedFileName: expectedFileName,
+        referenceDate: mockedCurrentDate,
+      });
+      expect(uploadedFileLog).toBe(true);
+    },
+  );
+
+  it(
+    "Should generate a SIMS to SFAS bridge file when there is an update on student assessment data " +
```

Comment:
The description of this test is exactly same as the next test. Let me know if I am reading it wrong.